### PR TITLE
nixos/deepin: init

### PR DESCRIPTION
**Pull request by wineee**
###### Description of changes

Adds the Deepin Desktop Environment to NixOS

Temporarily use kwin(plasma5) as window manager, but [deepin/dde-kwin](https://github.com/NixOS/nixpkgs/pull/215075)  is still  important for kwin effects

Test ON:
https://gist.github.com/wineee/eab3d40f9f60e37cfccf6c8ebe3502a7

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

**Comment by wineee** [comment]
@ofborg test deepin

File: nixos/modules/services/x11/desktop-managers/deepin.nix [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        default = [ ];
+        type = types.listOf types.path;
+        description = lib.mdDoc "List of packages for which gsettings are overridden.";
+      };
+    };
+
+    environment.deepin.excludePackages = mkOption {
+      default = [ ];
+      type = types.listOf types.package;
+      description = lib.mdDoc "Which packages pantheon should exclude from the default environment";
```
Comment (wegank) [inline nixos/modules/services/x11/desktop-managers/deepin.nix:34]: Pantheon?

Comment (wineee) [reply nixos/modules/services/x11/desktop-managers/deepin.nix]: a stupid mistake, fixed

**Comment by wegank** [comment]
Capture d’écran 2023-05-01 à 08 34 36

Excellent work!

**Review by wegank — APPROVED** [review]

**Pull request merged**